apps/web-giddh/src/app/accounting/journal-voucher/voucher/voucher.component.ts

```diff
@@ -675,12 +747,101 @@ export class AccountAsVoucherComponent implements OnInit, OnDestroy, AfterViewIn
 
         if (this.totalCreditAmount === this.totalDebitAmount) {
             if (this.validatePaymentAndReceipt(data)) {
+
+                let hasAdvanceReceipt = false;
+
+                if (this.requestObj.voucherType === VOUCHERS.RECEIPT) {
+
+                    this.validateEntries(true);
+
+                    if (!this.isValidForm) {
+                        return false;
+                    }
+
+                    let voucherAdjustments = this.receiptEntries;
+                    let totalAdjustmentAmount = 0;
+                    if (voucherAdjustments && voucherAdjustments.length > 0) {
+                        let dataVoucherAdjustments = [];
+                        let byEntry = data.transactions[1];
+                        let totalTransactions = data.transactions.length;
+                        let adjustmentsCount = 0;
+
+                        voucherAdjustments.forEach(adjustment => {
+                            totalAdjustmentAmount += Number(adjustment.amount);
+
+                            if (adjustment.type === "receipt" || adjustment.type === "advanceReceipt") {
+                                let taxAmount = 0;
+                                let advanceReceiptAmount = 0;
+
+                                if (adjustment.type === "advanceReceipt") {
+                                    hasAdvanceReceipt = true;
+                                    taxAmount = adjustment.tax.value;
+                                    advanceReceiptAmount = Number(adjustment.amount) - Number(taxAmount);
+                                }
+
+                                data.transactions[totalTransactions] = {
+                                    advanceReceiptAmount: advanceReceiptAmount,
+                                    amount: Number(adjustment.amount),
+                                    applyApplicableTaxes: byEntry.applyApplicableTaxes,
+                                    currentBalance: byEntry.applyApplicableTaxes,
+                                    discounts: [],
+                                    inventory: [],
+                                    isInclusiveTax: byEntry.isInclusiveTax,
+                                    particular: byEntry.particular,
+                                    selectedAccount: byEntry.selectedAccount,
+                                    stocks: null,
+                                    tax: taxAmount,
+                                    taxes: adjustment.tax.uniqueName ? [adjustment.tax.uniqueName] : [],
+                                    total: Number(adjustment.amount),
+                                    type: byEntry.type,
+                                    subVoucher: (adjustment.type === "advanceReceipt") ? "ADVANCE_RECEIPT" : ""
```

Comment:
Consider using `SubVoucher.AdvanceReceipt` enum as used in ledger and voucher module 
```suggestion
                                    subVoucher: (adjustment.type === "advanceReceipt") ? SubVoucher.AdvanceReceipt : ""
```